src/audio/generic_processor/generic_processor.c

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+
+/*
+ * A generic processor component.
+ */
+
+/**
+ * \file audio/generic_processor.c
+ * \brief Processing compoent aimed to work with external codecs
+ * \author Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ */
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/generic_processor/generic_processor.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/drivers/ipc.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/clk.h>
+#include <sof/lib/memory.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/ut.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <user/trace.h>
+
+static const struct comp_driver comp_generic_processor;
+
+//TODO: provide proper uuid below
+/* d8218443-5ff3-4a4c-b388-6cfe07b956aa */
+DECLARE_SOF_RT_UUID("generic_processor", gp_uuid, 0xd8218443, 0x5ff3, 0x4a4c,
+		    0xb3, 0x88, 0x6c, 0xfe, 0x07, 0xb9, 0x56, 0xaa);
+
+DECLARE_TR_CTX(gp_tr, SOF_UUID(gp_uuid), LOG_LEVEL_INFO);
+
+static struct comp_dev *generic_processor_new(const struct comp_driver *drv,
+					      struct sof_ipc_comp *comp)
+{
+	int ret;
+	struct comp_dev *dev = NULL;
+	struct comp_data *cd = NULL;
+	struct sof_ipc_comp_process *ipc_generic_processor =
+		(struct sof_ipc_comp_process *)comp;
+	struct generic_processor_config *cfg;
+	size_t bs;
+	void *lib_cfg;
+	size_t lib_cfg_size;
+
+	comp_cl_info(&comp_generic_processor, "generic_processor_new()");
+
+	dev = comp_alloc(drv, COMP_SIZE(struct sof_ipc_comp_process));
+	if (!dev) {
+		comp_cl_err(&comp_generic_processor, "generic_processor_new(), failed to allocate memory for comp_dev");
+		goto err;
+	}
+
+	dev->drv = drv;
+
+	ret = memcpy_s(&dev->comp, sizeof(struct sof_ipc_comp_process),
+		       comp, sizeof(struct sof_ipc_comp_process));
+	assert(!ret);
+
+	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
+	if (!cd) {
+		comp_cl_err(&comp_generic_processor, "generic_processor_new(), failed to allocate memory for comp_data");
+		goto err;
+	}
+
+	comp_set_drvdata(dev, cd);
+
+	/* Copy setup config */
+	cfg = (struct generic_processor_config *)ipc_generic_processor->data;
+	bs = ipc_generic_processor->size;
+	if (bs) {
+		if (bs < sizeof(struct generic_processor_config)) {
+			comp_info(dev, "generic_processor_new() error: wrong size of setup config");
+			goto err;
+		}
```

Comment:
Wrong trace function, btw. `ERROR` prefix is added to each `comp_err`/`tr_err` automatically, so there's no need to add them manually in message content.